src/rust/src/asn1.rs

```diff
@@ -77,21 +78,42 @@ struct DssSignature<'a> {
     s: asn1::BigUint<'a>,
 }
 
-pub(crate) fn big_asn1_uint_to_py<'p>(
+pub(crate) fn big_byte_slice_to_py_int<'p>(
     py: pyo3::Python<'p>,
-    v: asn1::BigUint<'_>,
+    v: &'_ [u8],
 ) -> pyo3::PyResult<&'p pyo3::PyAny> {
+    let signed = big_byte_slice_is_negative(py, v)?;
     let int_type = py.get_type::<pyo3::types::PyLong>();
-    int_type.call_method1("from_bytes", (v.as_bytes(), "big"))
+    let kwargs = [("signed", signed)].into_py_dict(py);
+    int_type.call_method("from_bytes", (v, "big"), Some(kwargs))
+}
+
+pub(crate) fn big_byte_slice_is_negative(
+    py: pyo3::Python<'_>,
+    bytes: &'_ [u8],
+) -> pyo3::PyResult<bool> {
+    if bytes[0] & 0x80 != 0 {
+        let cryptography_warning = py.import("cryptography.utils")?.getattr("DeprecatedIn36")?;
```

Comment:
This is not the right place for this warning. This function has an ostensibly generic purpose and instead it's got behavior that's totally specific to serials.